torchmultimodal/modules/layers/attention.py

```diff
@@ -0,0 +1,143 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+import itertools
+from typing import Optional
+
+import numpy as np
+
+import torch
+import torch.nn as nn
+
+from torchmultimodal.utils.common import tensor_slice
+
+
+class BroadcastPositionEmbedding(nn.Module):
+    r"""Spatiotemporal broadcasted positional embeddings.
+
+    Each embedding vector of the ``i``-th dim is repeated by ``N`` times, where
+    :math:`N = \prod_{j>i}\text{dim}[j]`.
+
+    Args:
+        shape (torch.Size): shape of raw data before batching and embedding
+        embedding_dim (int): the size of each embedding vector
+        dim (Optional[int]): dimension of embedding w.r.t. the other dimensions of batched data, e.g., ``-1``
+            corresponds to embedding being the last dimension. Default is ``-1``.
+
+    Raises:
+        ValueError: if ``dim`` is neither ``-1`` nor ``1``
+        ValueError: if ``embedding_dim`` is not an integer multiple of ``len(shape)``
+
+    Inputs:
+        x (Optional[torch.Tensor]): flattened input data, e.g., ``(batch, time * height * width, embedding_dim)``
+            if ``dim=-1``.
+        decode_step (Optional[int]): position of the data that requires decoding.
+    """
+
+    def __init__(
+        self, shape: torch.Size, embedding_dim: int, dim: Optional[int] = -1
+    ) -> None:
+        super().__init__()
+        if dim not in [-1, 1]:
```

Comment:
MUGEN has been ambivalent about [supporting both](https://github.com/mugen-org/MUGEN_baseline/blob/main/lib/models/gpt/attention.py#L259) or just `-1`. 
I removed the support for +1 to avoid this unnecessary complexity.